### PR TITLE
Rianfowler/sc 50084/ add and integrate /dashboard hook 

### PR DESCRIPTION
**Pull request by rianfowler**
#### What this PR does / why we need it:
- adds a hook for `/dashboard` fetching 
- integrates new hook into Dashboard component 

#### Which issue(s) this PR fixes:
https://app.shortcut.com/replicated/story/50084/fix-multi-app-ux

#### Does this PR introduce a user-facing change?
none- refactor 

#### Does this PR require documentation?
none

### CODE REVIEW

File: web/src/features/Dashboard/api/getSelectedAppClusterDashboard.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+export const useSelectedAppClusterDashboard = ({
+  refetchInterval = false,
+}: {
+  refetchInterval: number | false;
+}) => {
+  const { data: isHelmManagedResponse } = useIsHelmManaged();
+  const { isHelmManaged } = isHelmManagedResponse || { isHelmManaged: false };
+
+  const { selectedApp } = useSelectedApp();
+  const { slug } = selectedApp || { slug: "" };
```
Comment (rianfowler) [inline web/src/features/Dashboard/api/getSelectedAppClusterDashboard.ts:54]: This new hook composes other hooks for getting `isHelmManaged` and the `selectedApp`.

File: web/src/features/Dashboard/api/getSelectedAppClusterDashboard.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  const clusterId = selectedApp?.downstream?.cluster?.id.toString() || "";
+  return useQuery(
+    ["getAppClusterDashboared", slug, clusterId],
```
Comment (rianfowler) [inline web/src/features/Dashboard/api/getSelectedAppClusterDashboard.ts:57]: key the request off of `slug` and `clusterId`

File: web/src/features/Dashboard/api/useSelectedAppClusterDashboard.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      if (timerId.current) {
+        clearTimeout(timerId.current);
+      }
+    };
+  }, []);
+  // return everything that comes with react query
+  return { ...appClusterDashboardQuery, isSlowLoading };
+}
+
+export { useSelectedAppClusterDashboardWithIntercept };
```
Comment (rianfowler) [inline web/src/features/Dashboard/api/useSelectedAppClusterDashboard.ts:81]: Going to talk to Mia next week about our naming conventions. Just using this for now but I think we will look around at how BPR uses axios and try to follow their approach

File: web/src/features/Dashboard/components/Dashboard.tsx [hunk 4]
```diff
@@ -622,66 +623,34 @@ const Dashboard = (props: Props) => {
...
+      setState({
+        dashboard: {
+          appStatus: selectedAppClusterDashboardResponse.appStatus,
+          prometheusAddress:
+            selectedAppClusterDashboardResponse.prometheusAddress,
+          metrics: selectedAppClusterDashboardResponse.metrics,
+        },
+      });
+    }
+  }, [selectedAppClusterDashboardResponse]);
```
Comment (rianfowler) [inline web/src/features/Dashboard/components/Dashboard.tsx:640]: Pretty common pattern at this point:
- get rid of the fetching function
- replace with a custom hook that fetches
- set the state in `useEffect`

Eventually we will get rid of these `useEffects` and use the data directly but it's easier to use the existing state structure for this change for now

File: web/src/features/Dashboard/components/DashboardGraphsCard.tsx [hunk 5]
```diff
@@ -321,16 +278,6 @@ export default class DashboardGraphsCard extends React.Component<Props, State> {
...
-  componentDidMount() {
-    if (this.props.prometheusAddress) {
-      this.state.getAppDashboardJob.start(this.getAppDashboard, 2000);
-    }
-  }
-
-  componentWillUnmount() {
-    this.state.getAppDashboardJob.stop();
-  }
-
```
Comment (rianfowler) [inline web/src/features/Dashboard/components/DashboardGraphsCard.tsx:333]: as far as i can tell, this component doesn't use its dashboard state for anything. It renders the data passed in from its parent. 

I did take this concept and used it to add a `refetchInterval` to the dashboard call in `Dashboard.tsx`- so now the Dashboard refetches `/dashboard` on an interval and passes that data down to the `DashboardGraphsCard`

**Review by alicenstar — APPROVED** [review]

**Pull request merged**